api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -92,7 +114,14 @@ data SeverityNumber
   | Fatal3
   | Fatal4
   | Unknown !Int32
-  deriving (Eq, Ord, Read, Show)
+  deriving (Eq, Ord, Read, Show )
+
+instance Enum SeverityNumber where
+  toEnum 1 = Trace
+  toEnum 2 = Trace2
+  toEnum 3 = Trace3
+  toEnum 4 = Trace4
+  ... -}
```

Comment:
One possible way to represent `severityNumber` is to create an Enum containing the short names (e.g. Fatal2, Trace3) and use `fromEnum` to retrieve an `Int` when needed. The benefit of this approach is that it is simple to pattern match on the short names and to use a `Show` instance to convert them to `String`s. I am still undecided if this is the best way to represent `severityNumber`, so it is still an `Int64` for now. Another option would be to use pattern synonyms defined on a `newtype` of `Int64`.